roles/v2/mining-proxy/src/lib/downstream_mining.rs

```diff
@@ -342,62 +352,34 @@ impl
             .expect("No upstream initialized")
             .safe_lock(|s| s.channel_ids.safe_lock(|r| r.next()).unwrap())
             .unwrap();
-        let channel_kind = up
+        info!(channel_id);
+        if up
             .as_ref()
             .expect("No upstream initialized")
-            .safe_lock(|up| up.channel_kind)
-            .unwrap();
-        info!(channel_id);
-        match channel_kind {
-            ChannelKind::Group => Ok(SendTo::RelaySameMessageToRemote(up.unwrap())),
-            ChannelKind::Extended => {
-                let messages = up
-                    .as_mut()
-                    .unwrap()
-                    .safe_lock(|up| {
-                        up.open_standard_channel_down(
-                            req.request_id.as_u32(),
-                            req.nominal_hash_rate,
-                            true,
-                            channel_id,
-                        )
-                    })
-                    .unwrap();
-                for m in &messages {
-                    if let Mining::OpenStandardMiningChannelSuccess(m) = m {
-                        self.open_channel_for_down_hom_up_extended(
-                            m.channel_id,
-                            m.group_channel_id,
-                        );
-                    }
-                }
-                let messages = messages.into_iter().map(SendTo::Respond).collect();
-                Ok(SendTo::Multiple(messages))
-            }
-            ChannelKind::ExtendedWithNegotiator => {
-                let messages = up
-                    .as_mut()
-                    .unwrap()
-                    .safe_lock(|up| {
-                        up.open_standard_channel_down(
-                            req.request_id.as_u32(),
-                            req.nominal_hash_rate,
-                            true,
-                            channel_id,
-                        )
-                    })
-                    .unwrap();
-                for m in &messages {
-                    if let Mining::OpenStandardMiningChannelSuccess(m) = m {
-                        self.open_channel_for_down_hom_up_extended(
-                            m.channel_id,
-                            m.group_channel_id,
-                        );
-                    }
+            .safe_lock(|up| up.channel_kind.is_extended())
+            .unwrap()
+        {
+            let messages = up
+                .as_mut()
+                .unwrap()
+                .safe_lock(|up| {
```

Comment:
Instead of locking a 2nd time why not just pull the messages out in the above `safe_lock` call?

Reply:
I thought that it was more readable in that way and that the compiler would have been smart enough to not lock 2 times but I'm changing it to lock only one time.